GAE/src/org/akvo/flow/api/app/SurveyedLocaleServlet.java

```diff
@@ -0,0 +1,136 @@
+    private static final long serialVersionUID = 8748650927754433019L;
+    private SurveyedLocaleDao surveyedLocaleDao;
+    private static final Integer SL_PAGE_SIZE = 30;
+
+    public SurveyedLocaleServlet() {
+        setMode(JSON_MODE);
+        surveyedLocaleDao = new SurveyedLocaleDao();
+    }
+
+    @Override
+    protected RestRequest convertRequest() throws Exception {
+        HttpServletRequest req = getRequest();
+        RestRequest restRequest = new SurveyedLocaleRequest();
+        restRequest.populateFromHttpRequest(req);
+        return restRequest;
+    }
+
+    /**
+     * calls the surveyedLocaleDao to get the list of surveyedLocales for a certain surveyGroupId
+     * passed in via the request, or the total number of available surveyedLocales if the
+     * checkAvailable flag is set.
+     */
+    @Override
+    protected RestResponse handleRequest(RestRequest req) throws Exception {
+        SurveyedLocaleRequest slReq = (SurveyedLocaleRequest) req;
+        List<SurveyedLocale> slList;
+        if (slReq.getSurveyGroupId() != null) {
+            DeviceSurveyJobQueueDAO dsjqDAO = new DeviceSurveyJobQueueDAO();
+            SurveyDAO surveyDao = new SurveyDAO();
+            List<DeviceSurveyJobQueue> deviceSurveyJobQueues = dsjqDAO
+                    .get(slReq.getPhoneNumber(), slReq.getImei(), slReq.getAndroidId());
+            for (DeviceSurveyJobQueue dsjq : deviceSurveyJobQueues) {
+                Survey s = surveyDao.getById(dsjq.getSurveyID());
+                if (s != null && s.getSurveyGroupId().longValue()
+                        == slReq.getSurveyGroupId().longValue()) {
+                    slList = surveyedLocaleDao.listLocalesBySurveyGroupAndDate(
+                            slReq.getSurveyGroupId(), slReq.getLastUpdateTime(), SL_PAGE_SIZE);
+                    return convertToResponse(slList, slReq.getSurveyGroupId());
+                }
+            }
+        }
+        // A valid assignment has not been found for the given device
+        RestResponse res = new RestResponse();
+        res.setCode(String.valueOf(HttpServletResponse.SC_FORBIDDEN));
+        res.setMessage("Invalid assignment");
+        return res;
+    }
+
+    /**
+     * converts the domain objects to dtos and then installs them in a SurveyedLocaleResponse object
+     *
+     */
+    private SurveyedLocaleResponse convertToResponse(List<SurveyedLocale> slList, Long surveyId) {
+        SurveyedLocaleResponse resp = new SurveyedLocaleResponse();
+        if (slList == null) {
+            resp.setCode(String.valueOf(HttpServletResponse.SC_INTERNAL_SERVER_ERROR));
+            resp.setMessage("Internal Server Error");
+            return resp;
+        }
+        // set meta data
+        resp.setCode(String.valueOf(HttpServletResponse.SC_OK));
+        resp.setResultCount(slList.size());
+
+        List<SurveyedLocaleDto> dtoList = DataPointServlet.getSurveyedLocaleDtosList(slList, surveyId);
```

Comment:
In my opinion, there should be another class/classes containing methods used by both servlets. This will help have smaller classes and avoid static methods which are harder to test.

Reply:
Sure. Just hoped the old servlet would be removed soon. And why are static methods harder to test?

Reply:
When writing tests one usually initialize to class to test with a constructor but of course there are tools like powermock and then static classes methods can also be tested. So there could be a static utility class with utility methods but it seems unusual to me that a servlet calls another one. If you decide to remove the other servlet, then the methods would not need to be static any more. Also what disturbs me a bit is that DataPointServlet is used both with static methods and instantiated via constructor.

Reply:
ok to sum up: if they are only utility methods -> we can move them to a static utility class, I guess in this case this would require less rework and makes sense. _I still think they are not exactly utility classes since they access the database and read data and in this case if we wanted to mock some responses for them, they should be made into an object with constructor but it would require a change in architecture so maybe not needed right now._